### PR TITLE
Add support to build Host OS RPM

### PR DESCRIPTION
**Pull request by olavphibm**
The Host OS RPM makes it easier to setup Host OS repositories and start building supported packages.

There are two RPMs that are built:
 - host-os-repos: Makes sure centos-release and epel-release are installed and installs the Host OS .repo files.
 - host-os-devel: Installs the required packages to use the build scripts and installs an executable (host-os-restore-repos) that is able to restore the Host OS required .repo files in some cases.

This fixes open-power-host-os/builds#84.

There are many assumptions made here that I'm not sure are true:
- Should this RPM be included in our weekly repositories?
- Should the scripts build this RPM?
  - It could just be a static RPM available in some URL.
- Should the source files be included with the spec file?
  - We could have a separate repository from which the sources are downloaded.
- Which kind of errors should be repaired by the host-os-restore-repos.py script?
  - It currently needs to re-download the RPM files to restore the .repo from them. If the CentOS Base repository file is messed up and the repository is not cached, the script won't be able to repair the damage.

### CODE REVIEW

**Comment by fabianorosas** [comment]
Ok, here's my 2 cents:

>    host-os-repos: Makes sure centos-release and epel-release are installed and installs the Host OS .repo files.
  
I suggest host-os-release to keep it consistent.

>    host-os-devel: Installs the required packages to use the build scripts and installs an executable (host-os-restore-repos) that is able to restore the Host OS required .repo files in some cases.

Couldn't this be resolved by simply reinstalling the packages? What are these cases in which the script would be useful?

> There are many assumptions made here that I'm not sure are true:
> 
>    Should this RPM be included in our weekly repositories?

While this doesn't have much practical value since the user would have to download the .repo file to be able to get the package I think we could keep it in the repo for completion.

>    Should the scripts build this RPM?
>        It could just be a static RPM available in some URL.

I think it would make maintenance simple to just build the package alongside the others.

>    Should the source files be included with the spec file?
>        We could have a separate repository from which the sources are downloaded.

Do you mean the code from builds/versions? Perhaps in the hostos-devel package? The host-os-restore-repos script is not really a 'devel' tool.

>    Which kind of errors should be repaired by the host-os-restore-repos.py script?

I don't see any problems that could (frequently) happen with the repos. Perhaps make some general sanity checking to ensure we can run a build successfully? Even then, I think this should be in host-os-build.py

>    It currently needs to re-download the RPM files to restore the .repo from them. If the CentOS Base repository file is messed up and the repository is not cached, the script won't be able to repair the damage.

I see no point in having this level of recovery. Is there a specific scenario that we're trying to support?

**Comment by maurorodrigues** [comment]
> Should this RPM be included in our weekly repositories?
I don't think so. I fail to see a reason for that, this RPM makes sense externally, not for our test teams.

> Should the scripts build this RPM?
yes, maybe through a separated flag.

> It could just be a static RPM available in some URL.
Yes.
> Should the source files be included with the spec file?
Not sure I understand this one, I didn't review the patch yet, maybe I'll understand after it.

> We could have a separate repository from which the sources are downloaded.
> Which kind of errors should be repaired by the host-os-restore-repos.py script?
> It currently needs to re-download the RPM files to restore the .repo from them. If the CentOS Base repository file is messed up and the repository is not cached, the script won't be able to repair the damage.

**Comment by olavphibm** [comment]
>> Should the source files be included with the spec file?
>
> Not sure I understand this one, I didn't review the patch yet, maybe I'll understand after it.

I've included all the sources in the SOURCES directory, I felt no need to use a tar.gz file.
This needs a small modification in the builds scripts to allow building an RPM without archive, that I'll send if we follow this path.

**Comment by olavphibm** [comment]
> Ok, here's my 2 cents:
> 
>> host-os-repos: Makes sure centos-release and epel-release are installed and installs the Host OS .repo files.
> 
> I suggest host-os-release to keep it consistent.

That's fine for me.

> 
>> host-os-devel: Installs the required packages to use the build scripts and installs an executable (host-os-restore-repos) that is able to restore the Host OS required .repo files in some cases.
> 
> Couldn't this be resolved by simply reinstalling the packages? What are these cases in which the script would be useful?

Yes, it could, but it is not trivial to reinstall the centos-release package, because it is required by lots of other packages.
If the user messed up with his repo files though, I think he should be responsible for restoring them.

>> Should the source files be included with the spec file?
>> We could have a separate repository from which the sources are downloaded.
>
> Do you mean the code from builds/versions? Perhaps in the hostos-devel package? The host-os-restore-repos script is not really a 'devel' tool.

See my response to Mauro's comment above.

> I see no point in having this level of recovery. Is there a specific scenario that we're trying to support?

A scenario that was mentioned was that one of the required repositories is disabled. However, this is not the default state when the repositories are installed, and the user would have to modify the files to disable them. If we support this kind of thing, we should support "any" modifications that the user might make. I don't know how useful it is to support those scenarios.

File: host-os/centOS/7.2/SOURCES/host-os.repo [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[hostos]
+name=OpenPOWER Host OS
+baseurl=http://pokgsa.ibm.com/gsa/pokgsa/projects/h/host-os/weekly/latest/repository
```
Comment (mopsfelder) [inline host-os/centOS/7.2/SOURCES/host-os.repo]: This is an internal repository.

File: host-os/centOS/7.2/host-os.spec [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+%files repos
+%defattr(-,root,root,-)
+%doc GPLv3.txt
+%config(noreplace) /etc/yum.repos.d/*
+
+%files devel
+/usr/bin/host-os-restore-repos
+
+%changelog
```
Comment (mopsfelder) [inline host-os/centOS/7.2/host-os.spec]: It is missing a changelog entry here.

**Comment by olavphibm** [comment]
Temporarily closed

**Comment by olavphibm** [comment]
Since the initial requisites changed a lot, I'm closing this in favor of https://github.com/open-power-host-os/versions/pull/135

**Pull request closed without merging**